src/async-wrap.cc

```diff
@@ -282,37 +285,87 @@ bool AsyncWrap::EmitAfter(Environment* env, double async_id) {
 class PromiseWrap : public AsyncWrap {
  public:
   PromiseWrap(Environment* env, Local<Object> object, bool silent)
-    : AsyncWrap(env, object, PROVIDER_PROMISE, silent) {}
+    : AsyncWrap(env, object, PROVIDER_PROMISE, silent) {
+    MakeWeak(this);
```

Comment:
@addaleax , sorry for very basic question, why we add 
```c++
MakeWeak(this);
```
in this PR, since now we only keep the `asyncId` of parent promise here, is there any GC issue if we don't call `MakeWeak` ?